pkg/integration/handler.go

```diff
@@ -0,0 +1,340 @@
+package integration
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dhis2-sre/im-manager/pkg/config"
+
+	"github.com/dhis2-sre/im-manager/internal/handler"
+	"github.com/gin-gonic/gin"
+)
+
+type Handler struct {
+	config config.Config
+}
+
+func NewHandler(config config.Config) Handler {
+	return Handler{config}
+}
+
+type Request struct {
+	Key     string `json:"key" binding:"required"`
+	Payload any    `json:"payload"`
+}
+
+// Integrations ...
+// swagger:route POST /integrations postIntegration
+//
+// Return integration for a given key
+//
+// Security:
+//  oauth2:
+//
+// responses:
+//   200: Any
+//   401: Error
+//   403: Error
+//   415: Error
+func (h Handler) Integrations(c *gin.Context) {
+	var request Request
+	if err := handler.DataBinder(c, &request); err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	if request.Key == "IMAGE_REPOSITORY" {
+		token, err := getDockerHubToken(h.config.DockerHub.Username, h.config.DockerHub.Password)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		payload := request.Payload.(map[string]any)
+		organization := payload["organization"].(string)
+		images, err := getDockerHubImages(token, organization)
```

Comment:
I would create a docker hub client that is created once in our main using the config.
The client should then be passed to our handler or to our service which can call `getImages` or `getTags` on it.